brilliant-builders/firestarter/game_engine/object.py

```diff
@@ -0,0 +1,282 @@
+
+    def update(self, other_sprites: List[Sprite]) -> None:
+        pass
+
+    def on_collision(self, other: Sprite) -> bool:
+        if self.activated:
+            return self.collide
+
+        if isinstance(other, Player):
+            if other.checkpoint is not None:
+                print('deactivating')
+                other.checkpoint.deactivate()
+            self.activated = True
+            print("Checkpoint set!")
+            other.checkpoint = self
+
+        return self.collide
+
+    def deactivate(self) -> None:
+        self.activated = False
+        self.change_mode(2)
+
+
+class FlameBuddy(Sprite):
+    acc_x = NumericProperty(0)
+    acc_y = NumericProperty(0)
+    acc = ReferenceListProperty(acc_x, acc_y)
+    vel_x = NumericProperty(0)
+    vel_y = NumericProperty(0)
+    vel = ReferenceListProperty(vel_x, vel_y)
+
+    def __init__(
+            self,
+            sprite: Union[SpriteConfig, str],
+            pos: Tuple[int, int],
+            collide: bool = False,
+            mode: int = 1,
+            engine: Engine = None,
+            **kwargs
+    ):
+        # Resolve sprite if needed
+        if isinstance(sprite, str):
+            if not engine:
+                raise ValueError('Argument engine required when searching for sprite')
+            sprite = engine.assets[sprite]
+
+        super().__init__(sprite, pos, **kwargs)
+
+        self.activated = False
+        self.collide = collide
+        self.change_mode(mode)
+
+    def on_player_pos(self, new_pos: Tuple[float, float]) -> None:
+        distance_to_player = ((self.pos[0] - new_pos[0] + 10),
+                              (self.pos[1] - new_pos[1] - 55))
+
+        x_offset = -1 * distance_to_player[0] / 15
+
+        y_offset = -1 * distance_to_player[1] / 15
+
+        self.vel = (x_offset, y_offset)
+
+    def update(self, other_sprites: List[Sprite]) -> None:
+        self.vel = (self.vel_x + self.acc_x, self.vel_y + self.acc_y)
+        # self.vel = (min(self.vel_x, 5), min(self.vel_y, 5))
+        self.pos = (self.pos[0] + self.vel_x, self.pos[1] + self.vel_y)
+        self.acc = (0, 0)
+
+    def on_collision(self, other: Sprite) -> bool:
+        return self.collide
+
+
+class Player(Sprite):
+    acc_x = NumericProperty(0)
+    acc_y = NumericProperty(0)
+    acc = ReferenceListProperty(acc_x, acc_y)
+    vel_x = NumericProperty(0)
+    vel_y = NumericProperty(0)
+    vel = ReferenceListProperty(vel_x, vel_y)
+
+    score = NumericProperty(0)
+    lives = NumericProperty(5)
+
+    def __init__(
+            self,
+            config: SpriteConfig,
+            pos: Tuple[int, int] = (0, 0),
+            death_sound: WaveObject = None,
+            **kwargs
+    ) -> None:
+        super().__init__(config, pos, **kwargs)
+        self.is_standing: bool = False
+        self.checkpoint: Optional[Sprite] = None
+        self.respawn: Tuple[int, int] = pos
+
+        self.wall_boost = False
+
+        self.death_threshold = 0
```

Comment:
When assigning attributes like this, type checkers such as `mypy` will recognise the attribute as an integer. Later on in this file, you add a float to this attribute, so the attribute should be hinted accordingly:
```python
self.death_threshold: float = 0.0
# or:
self.death_threshold: Union[int, float] = 0